packages/libgcc_s1.rb

```diff
@@ -0,0 +1,355 @@
+      #system "python -O -m compileall tmp/#{CREW_DEST_PREFIX}/share/gcc-#{@gcc_version}"
+
+      ## Make symbolic links
+      #FileUtils.mkdir_p "#{CREW_DEST_LIB_PREFIX}/#{gcc_dir}"
+      #Dir.chdir "#{CREW_DEST_LIB_PREFIX}/#{gcc_dir}" do
+        #system "find . -type f -maxdepth 1 -exec ln -sv #{gcc_libdir}/{} #{CREW_DEST_LIB_PREFIX}/{} \\;"
+      #end
+
+      #Dir.chdir "#{CREW_DEST_MAN_PREFIX}/man1" do
+        #Dir.glob("*-#{gcc_version}.1*").each do |f|
+          #@basefile = f.gsub("-#{gcc_version}", '')
+          #FileUtils.ln_sf f, @basefile
+        #end
+      #end
+
+      ## Only make links to unversioned in postinstall
+      #Dir.chdir "tmp/#{CREW_DEST_PREFIX}/bin/" do
+        #Dir.glob("#{gcc_arch}-*-#{gcc_version}").each do |f|
+          ## @basefile_nover=f.split(/-#{gcc_version}/, 2).first
+          ## puts "Symlinking #{f} to #{@basefile_nover}"
+          ## FileUtils.ln_sf f, @basefile_nover
+          #@basefile_noarch = f.split(/#{gcc_arch}-/, 2).last
+          #puts "Symlinking #{f} to #{@basefile_noarch}"
+          #FileUtils.ln_sf f, @basefile_noarch
+        #end
+        ## Dir.glob("*-#{gcc_version}").each do |f|
+        ## @basefile_nover=f.split(/-#{gcc_version}/, 2).first
+        ## FileUtils.ln_sf f, @basefile_nover
+        ## end
+        ## # many packages expect this symlink
+        ## FileUtils.ln_sf "gcc-#{gcc_version}", 'cc'
+      #end
+      ## libgomp.so conflicts with llvm
+      #@deletefiles = %W[#{CREW_DEST_LIB_PREFIX}/libgomp.so]
+      #@deletefiles.each do |f|
+        #FileUtils.rm f if File.exist?(f)
+      #end
+    #end
+  #end
+
+  def self.postinstall
+  # built using this:
+    ##!/bin/bash -e
+    ## Usage: ./extracttar original_tar_archive new_tar_archive files_or_pattern_to_extract
+    ## One may pass the --wildcards flag if necessary.
+    #origfile=$1
+    #newfile=$2
+    #startdir="$(pwd)"
+    #extract="${@:3}"
+    #sanitized_extract=
+    #for i in $extract
+    #do
+    #[[ $i == /* ]] && sanitized_extract+=" ${i#/} "
+    #done
+    
+    #tmpdir=$(mktemp -d)
+    #mkdir -p $tmpdir/extract_dir
+    #cd $tmpdir/extract_dir && tar afvx "$startdir"/$origfile $(echo "$sanitized_extract")
+    ## create file list
+    #find . -type f > ../filelist
+    #find . -type l >> ../filelist
+    #cut -c2- ../filelist > filelist
+    ## create directory list
+    #find . -type d > ../dlist
+    #cut -c2- ../dlist > dlistcut
+    #tail -n +2 dlistcut > dlist
+    ## Create new archive
+    #[[ -e "$startdir"/${newfile} ]] && mv  "$startdir"/${newfile} "$startdir"/${newfile}.old
+    #tar -c * --owner=chronos:1000 --group=chronos:1000 --use-compress-program='xz -9 -T 0' -f "$startdir"/${newfile}
+    #cd "$startdir"
+    #rm -rf $tmpdir
+    #sha256sum ${newfile} > ${newfile}.sha256
+  # extracttar gcc11-11.1.0-chromeos-x86_64.tar.xz libgcc_s1-11.1.0-chromeos-x86_64.tar.xz /usr/local/lib64/libgcc_s.so.1
+  end
```

Comment:
This is the big question. I'm using the script commented out in `postinstall` there, to extract the files from the package built for gcc11. 

Essentially:
```
#!/bin/bash -e
# Usage: ./extracttar original_tar_archive new_tar_archive files_or_pattern_to_extract
# One may pass the --wildcards flag if necessary.
origfile=$1
newfile=$2
startdir="$(pwd)"
extract="${@:3}"
sanitized_extract=
for i in $extract
do
[[ $i == /* ]] && sanitized_extract+=" ${i#/} "
done

tmpdir=$(mktemp -d)
mkdir -p $tmpdir/extract_dir
cd $tmpdir/extract_dir && tar afvx "$startdir"/$origfile $(echo "$sanitized_extract")
echo "workdir is $(pwd)"
# create file list
find . -type f > ../filelist
find . -type l >> ../filelist
cut -c2- ../filelist > filelist
# create directory list
find . -type d > ../dlist
cut -c2- ../dlist > dlistcut
tail -n +2 dlistcut > dlist
# Create new archive
[[ -e "$startdir"/${newfile} ]] && mv  "$startdir"/${newfile} "$startdir"/${newfile}.old
tar -c * --owner=chronos:1000 --group=chronos:1000 --use-compress-program='xz -9 -T 0' -f "$startdir"/${newfile}
cd "$startdir"
rm -rf $tmpdir
sha256sum ${newfile} > ${newfile}.sha256
```

We need a way of using parent builds to create multiple packages. This is just a hacky workaround, to avoid having to install massive packages...

Any suggestions?

Reply:
How do other distros like Debian or Gentoo split it out?

Reply:
They all split build and packaging into distinct steps, with source package mapping to multiple child binary packages through separate packaging steps per child binary package.

Reply:
e.g. https://github.com/archlinux/svntogit-packages/blob/packages/gcc/trunk/PKGBUILD
Debian is more complicated: https://salsa.debian.org/toolchain-team/gcc/-/tree/master/debian

Reply:
Also, this is dependent upon llvm 12, so I'm dropping this back down to draft until I get the `libLLVM-12.so` broken out so there isn't a 1.4Gb dependency on llvm.

Reply:
This all makes my head spin.  You like to climb the steepest mountains, apparently.  LoL